business-services/dashboard-analytics/src/main/java/com/tarento/analytics/handler/ReportChartResponseHandler.java

```diff
@@ -0,0 +1,175 @@
+                String path = jsonPathObj.get("path").asText();
+                String dataType = jsonPathObj.get("dataType").asText();
+                String columnName = jsonPathObj.get("columnName").asText();
+                String val = null;
+                try {
+                    val = JsonPath.read(sourceData.toString(), "$."+path).toString();
+                } catch (PathNotFoundException e) {
+                    val = null;
+                }
+                if (dataType.equals(STRING_DATATYPE)) {
+                    Plot plot = new Plot(columnName, val==null? val: String.valueOf(val), STRING_DATATYPE);
+                    plotList.add(plot);
+
+                } else {
+                    Plot plot = new Plot(columnName, val==null? null : Double.valueOf(val), dataType);
+                    plotList.add(plot);
+                }
+            });
+            Data data = new Data(String.valueOf(srNo), srNo, null);
+            data.setPlots(plotList);
+            dataList.add(data);
+        });
+        AggregateDto aggregateDto= getAggregatedDto(chartNode, dataList, requestDto.getVisualizationCode());
+//        aggregateDto.setTotal(total);
+        return aggregateDto;
+    }
+
+    private AggregateDto getAggregationData(AggregateRequestDto requestDto, ObjectNode aggregations) {
+        JsonNode aggregationNode = aggregations.get(AGGREGATIONS);
+        JsonNode chartNode = requestDto.getChartNode();
+        ArrayNode pathDataTypeMap = (ArrayNode) chartNode.get(TYPE_MAPPING);
+        Map<String, String> dataTypeMap = new HashMap<String, String>();
+        pathDataTypeMap.forEach(type -> {
+            String key = type.fields().next().getKey();
+            JsonNode value = type.fields().next().getValue();
+            dataTypeMap.put(key, value.asText());
+        });
+        ArrayNode aggrsPaths = (ArrayNode) chartNode.get(IResponseHandler.AGGS_PATH);
+        List<Data> dataList = new ArrayList<>();
+        if (!aggrsPaths.isEmpty()) {
+            List<JsonNode> aggrNodes = aggregationNode.findValues(BUCKETS);
+            Map<String, String> levelValues = new HashMap<String, String>();
+            aggrNodes.stream().forEach(node -> {
+                ArrayNode buckets = (ArrayNode) node;
+                buckets.forEach(bucket -> {
+                    getAggregationBuckets(bucket, aggrsPaths, 0, dataList, levelValues, dataTypeMap);
+                });
+            });
+        }
+
+        return getAggregatedDto(chartNode, dataList, requestDto.getVisualizationCode());
+    }
+
+    private void getAggregationBuckets(JsonNode aggregationNode, ArrayNode aggrsPaths, int idx, List<Data> dataList, Map<String, String> levelValues, Map<String, String> dataTypeMap) {
+        //
+        levelValues.put(aggrsPaths.get(idx).asText(), aggregationNode.get(KEY).asText());
+
+        List<JsonNode> aggrNodes = aggregationNode.findValues(BUCKETS);
+        if (aggrNodes.isEmpty()) {
```

Comment:
use CollectionUtils.isEmpty(aggrNodes) instead of invoking isEmpty() on aggrNodes list as it is not null safe.